<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>1200</width>
    <height>700</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>MainWindow</string>
  </property>
  <widget class="QWidget" name="centralwidget">
   <widget class="QGraphicsView" name="graphicsView">
    <property name="geometry">
     <rect>
      <x>0</x>
      <y>0</y>
      <width>1191</width>
      <height>701</height>
     </rect>
    </property>
    <property name="dragMode">
     <enum>QGraphicsView::ScrollHandDrag</enum>
    </property>
   </widget>
   <widget class="QPushButton" name="PlusButton">
    <property name="geometry">
     <rect>
      <x>40</x>
      <y>20</y>
      <width>41</width>
      <height>41</height>
     </rect>
    </property>
    <property name="text">
     <string>➕</string>
    </property>
   </widget>
   <widget class="QPushButton" name="MinusButton">
    <property name="geometry">
     <rect>
      <x>100</x>
      <y>20</y>
      <width>41</width>
      <height>41</height>
     </rect>
    </property>
    <property name="text">
     <string>➖</string>
    </property>
   </widget>
   <widget class="QPushButton" name="OnOffButton">
    <property name="geometry">
     <rect>
      <x>170</x>
      <y>20</y>
      <width>80</width>
      <height>31</height>
     </rect>
    </property>
    <property name="styleSheet">
     <string notr="true">QPushButton { background-color: rgb(255, 132, 139); border-radius: 3px; color: rgb(255, 255, 255); } QPushButton:hover { background-color: rgb(245, 69, 234); }</string>
    </property>
    <property name="text">
     <string/>
    </property>
    <property name="icon">
     <iconset theme="quit" resource="map.qrc">
      <normaloff>:/new/exit.png</normaloff>:/new/exit.png</iconset>
    </property>
    <property name="iconSize">
     <size>
      <width>40</width>
      <height>40</height>
     </size>
    </property>
   </widget>
   <widget class="QPushButton" name="LeftButton">
    <property name="geometry">
     <rect>
      <x>1000</x>
      <y>610</y>
      <width>61</width>
      <height>31</height>
     </rect>
    </property>
    <property name="font">
     <font>
      <pointsize>18</pointsize>
     </font>
    </property>
    <property name="text">
     <string>←</string>
    </property>
   </widget>
   <widget class="QPushButton" name="RightButton">
    <property name="geometry">
     <rect>
      <x>1090</x>
      <y>610</y>
      <width>61</width>
      <height>31</height>
     </rect>
    </property>
    <property name="font">
     <font>
      <pointsize>18</pointsize>
     </font>
    </property>
    <property name="text">
     <string>→</string>
    </property>
   </widget>
   <widget class="QPushButton" name="ClearButton">
    <property name="geometry">
     <rect>
      <x>460</x>
      <y>20</y>
      <width>51</width>
      <height>51</height>
     </rect>
    </property>
    <property name="text">
     <string/>
    </property>
    <property name="icon">
     <iconset theme="reload" resource="map.qrc">
      <normaloff>:/new/reload.png</normaloff>:/new/reload.png</iconset>
    </property>
    <property name="iconSize">
     <size>
      <width>45</width>
      <height>45</height>
     </size>
    </property>
   </widget>
   <widget class="QPushButton" name="dijkstras">
    <property name="geometry">
     <rect>
      <x>170</x>
      <y>70</y>
      <width>81</width>
      <height>31</height>
     </rect>
    </property>
    <property name="styleSheet">
     <string notr="true">QPushButton {
    background-color: rgb(85, 170, 85); /* 浅绿色 */
    border-radius: 3px;
    color: rgb(255, 255, 255);
}

QPushButton:hover {
    background-color: rgb(0, 128, 0); /* 深绿色 */
}

QPushButton:pressed {
    background-color: rgb(0, 100, 0); /* 更深的绿色 */
}
</string>
    </property>
    <property name="text">
     <string>dijkstras</string>
    </property>
   </widget>
   <widget class="QLabel" name="label">
    <property name="geometry">
     <rect>
      <x>270</x>
      <y>20</y>
      <width>191</width>
      <height>41</height>
     </rect>
    </property>
    <property name="text">
     <string>TextLabel</string>
    </property>
   </widget>
   <widget class="QLabel" name="label_2">
    <property name="geometry">
     <rect>
      <x>270</x>
      <y>70</y>
      <width>171</width>
      <height>31</height>
     </rect>
    </property>
    <property name="text">
     <string>TextLabel</string>
    </property>
   </widget>
   <widget class="QPushButton" name="Astar">
    <property name="geometry">
     <rect>
      <x>170</x>
      <y>100</y>
      <width>81</width>
      <height>31</height>
     </rect>
    </property>
    <property name="styleSheet">
     <string notr="true">QPushButton {
    background-color: rgb(65, 105, 225); /* 蓝色 */
    border-radius: 3px;
    color: rgb(255, 255, 255);
}

QPushButton:hover {
    background-color: rgb(30, 144, 255); /* 深蓝色 */
}

QPushButton:pressed {
    background-color: rgb(0, 0, 205); /* 更深的蓝色 */
}
</string>
    </property>
    <property name="text">
     <string>A*</string>
    </property>
   </widget>
   <widget class="QLabel" name="label_3">
    <property name="geometry">
     <rect>
      <x>270</x>
      <y>100</y>
      <width>171</width>
      <height>31</height>
     </rect>
    </property>
    <property name="text">
     <string>TextLabel</string>
    </property>
   </widget>
   <widget class="QPushButton" name="CompareButton">
    <property name="geometry">
     <rect>
      <x>460</x>
      <y>80</y>
      <width>51</width>
      <height>51</height>
     </rect>
    </property>
    <property name="text">
     <string/>
    </property>
    <property name="icon">
     <iconset theme="kzzz" resource="map.qrc">
      <normaloff>:/new/pen.png</normaloff>:/new/pen.png</iconset>
    </property>
    <property name="iconSize">
     <size>
      <width>45</width>
      <height>45</height>
     </size>
    </property>
   </widget>
   <widget class="QDoubleSpinBox" name="doubleSpinBox">
    <property name="geometry">
     <rect>
      <x>40</x>
      <y>80</y>
      <width>101</width>
      <height>31</height>
     </rect>
    </property>
    <property name="value">
     <double>1.000000000000000</double>
    </property>
   </widget>
   <widget class="QGraphicsView" name="graphicsView_2">
    <property name="geometry">
     <rect>
      <x>790</x>
      <y>10</y>
      <width>381</width>
      <height>261</height>
     </rect>
    </property>
   </widget>
   <widget class="QCheckBox" name="Text_On_Button">
    <property name="geometry">
     <rect>
      <x>40</x>
      <y>120</y>
      <width>61</width>
      <height>18</height>
     </rect>
    </property>
    <property name="text">
     <string>text on</string>
    </property>
   </widget>
   <widget class="QPushButton" name="home">
    <property name="geometry">
     <rect>
      <x>560</x>
      <y>20</y>
      <width>51</width>
      <height>51</height>
     </rect>
    </property>
    <property name="text">
     <string/>
    </property>
    <property name="icon">
     <iconset resource="map.qrc">
      <normaloff>:/new/home.jpg</normaloff>:/new/home.jpg</iconset>
    </property>
    <property name="iconSize">
     <size>
      <width>45</width>
      <height>45</height>
     </size>
    </property>
    <property name="autoRepeat">
     <bool>false</bool>
    </property>
   </widget>
   <widget class="QPushButton" name="mhd">
    <property name="geometry">
     <rect>
      <x>620</x>
      <y>20</y>
      <width>51</width>
      <height>51</height>
     </rect>
    </property>
    <property name="text">
     <string/>
    </property>
    <property name="icon">
     <iconset theme="kzz" resource="map.qrc">
      <normaloff>:/new/M.png</normaloff>:/new/M.png</iconset>
    </property>
    <property name="iconSize">
     <size>
      <width>45</width>
      <height>45</height>
     </size>
    </property>
   </widget>
   <widget class="QPushButton" name="obstacle">
    <property name="geometry">
     <rect>
      <x>560</x>
      <y>80</y>
      <width>51</width>
      <height>51</height>
     </rect>
    </property>
    <property name="text">
     <string/>
    </property>
    <property name="icon">
     <iconset theme="kzz" resource="map.qrc">
      <normaloff>:/new/object.png</normaloff>
      <selectedon>:/new/object.png</selectedon>:/new/object.png</iconset>
    </property>
    <property name="iconSize">
     <size>
      <width>45</width>
      <height>45</height>
     </size>
    </property>
   </widget>
   <widget class="QPushButton" name="darkmode">
    <property name="geometry">
     <rect>
      <x>620</x>
      <y>80</y>
      <width>51</width>
      <height>51</height>
     </rect>
    </property>
    <property name="text">
     <string/>
    </property>
    <property name="icon">
     <iconset theme="kzz" resource="map.qrc">
      <normaloff>:/new/light.png</normaloff>:/new/light.png</iconset>
    </property>
    <property name="iconSize">
     <size>
      <width>45</width>
      <height>45</height>
     </size>
    </property>
   </widget>
   <widget class="QPushButton" name="walk">
    <property name="geometry">
     <rect>
      <x>680</x>
      <y>20</y>
      <width>51</width>
      <height>51</height>
     </rect>
    </property>
    <property name="text">
     <string/>
    </property>
    <property name="icon">
     <iconset theme="hospital" resource="map.qrc">
      <normaloff>:/new/walk.png</normaloff>:/new/walk.png</iconset>
    </property>
    <property name="iconSize">
     <size>
      <width>45</width>
      <height>45</height>
     </size>
    </property>
   </widget>
   <widget class="QPushButton" name="priority">
    <property name="geometry">
     <rect>
      <x>680</x>
      <y>80</y>
      <width>51</width>
      <height>51</height>
     </rect>
    </property>
    <property name="text">
     <string/>
    </property>
    <property name="icon">
     <iconset resource="map.qrc">
      <normaloff>:/new/road.png</normaloff>:/new/road.png</iconset>
    </property>
    <property name="iconSize">
     <size>
      <width>45</width>
      <height>45</height>
     </size>
    </property>
   </widget>
  </widget>
  <widget class="QMenuBar" name="menubar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>1200</width>
     <height>17</height>
    </rect>
   </property>
  </widget>
  <widget class="QStatusBar" name="statusbar"/>
 </widget>
 <resources>
  <include location="map.qrc"/>
 </resources>
 <connections>
  <connection>
   <sender>OnOffButton</sender>
   <signal>clicked()</signal>
   <receiver>MainWindow</receiver>
   <slot>close()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>209</x>
     <y>52</y>
    </hint>
    <hint type="destinationlabel">
     <x>333</x>
     <y>237</y>
    </hint>
   </hints>
  </connection>
 </connections>
</ui>
